java-frontend/src/main/java/org/sonar/java/model/DefaultJavaFileScannerContext.java

```diff
@@ -261,4 +262,9 @@ protected static void throwIfEndOfAnalysisCheck(JavaCheck javaCheck) {
     }
     return Optional.empty();
   }
+
+  @Override
+  public JavaIssueBuilder newIssue() {
+    return new InternalJavaIssueBuilder(inputFile, sonarComponents);
```

Comment:
I wonder if we should rework the various `reportIssue()` methods to use the new API...

Reply:
I would suggest considering doing this as a second step, this PR is getting quite big, and I would rather not add too much to it.